platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/network/connectivity/SocketFactoryTest.java

```diff
@@ -67,38 +78,48 @@ class SocketFactoryTest {
      *
-     * @param socketFactory1
-     * 		a factory for both server and client sockets
-     * @param socketFactory2
-     * 		a factory for both server and client sockets
+     * @param socketFactory1 a factory for both server and client sockets
+     * @param socketFactory2 a factory for both server and client sockets
+     * @param peerInfoList a list of peers
      */
-    private static void testSocketsBoth(final SocketFactory socketFactory1, final SocketFactory socketFactory2)
+    private static void testSocketsBoth(
+            final SocketFactory socketFactory1, final SocketFactory socketFactory2, final List<PeerInfo> peerInfoList)
             throws Throwable {
-        testSockets(socketFactory1, socketFactory2);
-        testSockets(socketFactory2, socketFactory1);
+        testSockets(socketFactory1, socketFactory2, peerInfoList);
+        testSockets(socketFactory2, socketFactory1, peerInfoList);
     }
 
     /**
-     * - establishes a connection using the provided factories
-     * - transfers some data
-     * - verifies the transferred data is correct
-     * - closes the sockets
+     * - establishes a connection using the provided factories - transfers some data - verifies the transferred data is
+     * correct - closes the sockets
      *
-     * @param serverFactory
-     * 		factory to create the server socket
-     * @param clientFactory
-     * 		factory to create the client socket
+     * @param serverFactory factory to create the server socket
+     * @param clientFactory factory to create the client socket
+     * @param peerInfoList a list of peers
      */
-    private static void testSockets(final SocketFactory serverFactory, final SocketFactory clientFactory)
+    private static void testSockets(
+            final SocketFactory serverFactory, final SocketFactory clientFactory, final List<PeerInfo> peerInfoList)
             throws Throwable {
 
         final ServerSocket serverSocket = serverFactory.createServerSocket(PORT);
 
         final Thread server = new Thread(() -> {
             try {
                 final Socket s = serverSocket.accept();
+                if (s instanceof SSLSocket) {
+                    if (peerInfoList.stream().anyMatch(peer -> mockingDetails(peer.signingCertificate())
+                            .isMock())) {
+                        // we've passed a mocked peer's certificate as an example of an invalid agreement
+                        // certificate - it possesses no valid issuer's principal
+                        assertNull(NetworkUtils.identifyTlsPeer(
+                                ((SSLSocket) s).getSession().getPeerCertificates(), peerInfoList));
+                    } else {
+                        assertNotNull(NetworkUtils.identifyTlsPeer(
+                                ((SSLSocket) s).getSession().getPeerCertificates(), peerInfoList));
+                    }
+                }
```

Comment:
this looks like its beyond the scope of this test

Reply:
moved out to a dedicated test